salt/utils/aws.py

```diff
@@ -89,9 +89,7 @@ def creds(provider):
                 proxies={'http': ''}, timeout=AWS_METADATA_TIMEOUT,
             )
             result.raise_for_status()
-            role = result.text.encode(
-                result.encoding if result.encoding else 'utf-8'
-            )
+            role = result.text
```

Comment:
This is going to need more testing, because the result should report back the encoding, and it is used in other places fine.

They may be better to just run `salt.utils.stringutils.to_str` for places where it is needed, like 

```
"http://169.254.169.254/latest/meta-data/iam/security-credentials/{0}".format(salt.utils.stringutils.to_str(role)),
```